datajoint/erd.py

```diff
@@ -1,473 +1,272 @@
-        2) if i^th nodes are at the same depth, i^th node in path 1 is alphabetically less than i^th node
-        in path 2
-        3) if neither of the above statement is true even if path1 and path2 are switched, proceed to i+1^th node
-        If path2 is a subpath start at node 1, then path1 is greater than path2
-        :param path1: path 1 of 2 to be compared
-        :param path2: path 2 of 2 to be compared
-        :return: return 1 if path1 is greater than path2, -1 if path1 is less than path2, and 0 if they are identical
-        """
-        node_depth_lookup = dict(self.nodes_by_depth())
-        for node1, node2 in zip(path1, path2):
-            if node_depth_lookup[node1] != node_depth_lookup[node2]:
-                return -1 if node_depth_lookup[node1] < node_depth_lookup[node2] else 1
-            if node1 != node2:
-                return -1 if node1 < node2 else 1
-        if len(node1) != len(node2):
-            return -1 if len(node1) < len(node2) else 1
-        return 0
-
-    def repr_path_with_depth(self, path, n=20, m=2):
-        node_depth_lookup = dict(self.nodes_by_depth())
-        node_labels = self.node_labels
-        space = '-' * n
-        rep = ''
-        prev_depth = 0
-        first = True
-        for (i, node) in enumerate(path):
-            depth = node_depth_lookup[node]
-            label = node_labels[node]
-            if first:
-                rep += (' '*(n+m))*(depth-prev_depth)
-            else:
-                rep += space.join(['-'*m]*(depth-prev_depth))[:-1] + '>'
-            first = False
-            prev_depth = depth
-            if i == len(path)-1:
-                rep += label
-            else:
-                rep += label.ljust(n, '-')
-        rep += '\n'
-        return rep
-
-    @classmethod
-    def create_from_dependencies(cls, dependencies, *args, **kwargs):
-        obj = cls(*args, **kwargs)
-
-        for full_table, parents in dependencies.parents.items():
-            database, table = (x.strip('`') for x in full_table.split('.'))
-            obj.add_node(full_table, database=database, table=table)
-            for parent in parents:
-                obj.add_edge(parent, full_table, rel='parent')
-
-        # create non primary key foreign connections
-        for full_table, referenced in dependencies.referenced.items():
-            for ref in referenced:
-                obj.add_edge(ref, full_table, rel='referenced')
-
-        return obj
+        graph = self._make_graph(prefix_module)
+        if pos is None:
+            pos = (layout if layout else self._layout)(graph, **layout_options)
+        import matplotlib.pyplot as plt
+
+        edge_list = graph.edges(data=True)
+        edge_styles = ['solid' if e[2]['primary'] else 'dashed' for e in edge_list]
+        nx.draw_networkx_edges(graph, pos=pos, edgelist=edge_list, style=edge_styles, alpha=0.2)
+
+        label_props = { # http://matplotlib.org/examples/color/named_colors.html
```

Comment:
That could go up on module level so because it

- does not need to be defined every time
- can be modified by the user there